referral.go

```diff
@@ -187,3 +196,78 @@ func (c *Client) createEasypostCreditCard(ctx context.Context, referralCustomerA
 }
+
+// BetaAddPaymentMethod adds Stripe payment method to referral customer.
+func (c *Client) BetaAddPaymentMethod(stripeCustomerId string, paymentMethodReference string) (out *PaymentMethodObject, err error) {
+	return c.BetaAddPaymentMethodWithContext(context.Background(), stripeCustomerId, paymentMethodReference)
+}
+
+// BetaAddPaymentMethodWithContext performs the same operation as BetaAddPaymentMethod, but allows
+// specifying a context that can interrupt the request.
+func (c *Client) BetaAddPaymentMethodWithContext(ctx context.Context, stripeCustomerId string, paymentMethodReference string) (out *PaymentMethodObject, err error) {
+	return c.BetaAddPaymentMethodWithPrimaryOrSecondaryAndContext(ctx, stripeCustomerId, paymentMethodReference, PrimaryPaymentMethodPriority)
+}
+
+// BetaAddPaymentMethodWithPrimaryOrSecondary adds Stripe payment method to referral customer with
+// PaymentMethodPriority parameter.
+func (c *Client) BetaAddPaymentMethodWithPrimaryOrSecondary(stripeCustomerId string, paymentMethodReference string, primaryOrSecondary PaymentMethodPriority) (out *PaymentMethodObject, err error) {
+	return c.BetaAddPaymentMethodWithPrimaryOrSecondaryAndContext(context.Background(), stripeCustomerId, paymentMethodReference, primaryOrSecondary)
+}
+
+// BetaAddPaymentMethodWithPrimaryOrSecondaryAndContext performs the same operation as BetaAddPaymentMethodWithPrimaryOrSecondary, but allows
+// specifying a context that can interrupt the request.
+func (c *Client) BetaAddPaymentMethodWithPrimaryOrSecondaryAndContext(ctx context.Context, stripeCustomerId string, paymentMethodReference string, primaryOrSecondary PaymentMethodPriority) (out *PaymentMethodObject, err error) {
+	primaryOrSecondaryEndpoint := ""
+
+	switch primaryOrSecondary {
+	case PrimaryPaymentMethodPriority:
+		primaryOrSecondaryEndpoint = "primary"
+	case SecondaryPaymentMethodPriority:
+		primaryOrSecondaryEndpoint = "secondary"
+	}
+
+	params := map[string]interface{}{
+		"stripe_customer_id":       stripeCustomerId,
+		"payment_method_reference": paymentMethodReference,
+		"priority":                 primaryOrSecondaryEndpoint,
+	}
+
+	wrappedParams := map[string]interface{}{
+		"payment_method": params,
+	}
+
+	err = c.post(ctx, "/beta/referral_customers/payment_method", wrappedParams, out)
+	return
+}
+
+// BetaRefundByAmount refunds a recent payment by amount in cents.
+func (c *Client) BetaRefundByAmount(refundAmount int) (out *BetaPaymentRefund, err error) {
+	return c.BetaRefundByAmountWithContext(context.Background(), refundAmount)
+}
+
+// BetaRefundByAmountWithContext performs the same operation as BetaRefundByAmount, but allows
+// specifying a context that can interrupt the request.
+func (c *Client) BetaRefundByAmountWithContext(ctx context.Context, refundAmount int) (out *BetaPaymentRefund, err error) {
+	params := map[string]interface{}{
+		"refund_amount": refundAmount,
+	}
+
+	err = c.post(ctx, "/beta/referral_customers/refunds", params, out)
```

Comment:
Question: Is this how we do beta endpoints in Go, just add "beta" to the endpoint? No bool or enum that triggers the API version to change? So all other post functions in here say `/v2/something/something`?

Reply:
Yes, if you pass in the slash in front of the endpoint, it will override the `v2` endpoint. You don't need to append `/v2` in other production functions, we have declared the API version path here: https://github.com/EasyPost/easypost-go/blob/master/client.go#L20

Reply:
Thanks, I hate this anti-pattern 🙃